_posts/2017/2017-11-01-internal-workshops.md

```diff
@@ -0,0 +1,197 @@
+
+A recent internal reorganization was a good occasion to revise and update the content
+of the workshop, and present it again to interested colleagues.
+
+
+## Why (not) Gradle?
+
+Here is a [good article by Benjamin Muschko](http://www.drdobbs.com/jvm/why-build-your-java-projects-with-gradle/240168608)
+that explains the evolution of Java build tools.<br>
+The comments, especially the first one by "phansson", are also worth reading, because
+instead of discussing pros and cons one feature or another, they address a more
+fundamental question: Is it really worth it?
+
+Gradle is probably not what you would consider a lightweight tool. The user guide
+has 700 pages and the API documentation contains about 1100 classes.<br>
+It is definitely intimidating, which means that the intention to use it should be
+well justified. Especially in today's working environments, where the price of accidental
+complexity is already very high, and in the face of [alternatives](https://maven.apache.org/guides/index.html),
+which are arguably simpler and easier to get started with.
+
+I have seen quite a few Gradle projects so far, in various companies of different
+sizes.<br>
+Except for a couple insignificant ones, mostly all of them were [multi-projects](https://docs.gradle.org/current/userguide/multi_project_builds.html).
+According to the documentation, "this topic is also the most intellectually challenging".<br>
+I find this hint very peculiar, and I consider it a warning.
+
+One project I had worked with consisted of about 60 (Java-)subprojects.<br>
+It did have this problem when you imported the project(s) into Eclipse (after running
+`./gradlew eclipse`), it would go into an apparently endless loop of opening projects.<br>
+There was no problem with the build according to Gradle, but Eclipse somehow
+didn't like the structure. The workaround was to cancel the import operation, and
+trigger a "Clean" operation in Eclipse.
+Considering this needed to happen quite often (basically every time a developer
+changed his branch), this was quickly getting annoying.<br>
+
+Another problem was with the database configuration. There existed a configuration
+file, but *it was not used during (integration-) testing*. Instead, a subproject
+called `datasource-test` was created, which was pulled in by another subproject
+specifying a `testRuntime` dependency on it.<br>
+(The subprojects did have a tree-structure, but were located on the same directory
+level...)<br>
+The `datasource-test` project then used dependency injection to bind the database
+configuration to a class, that had *hardcoded* values in it.<br>
+Basically, there was a bunch of highly convoluted Gradle and Guice code, that
+did nothing else besides *reimplementing something that was already configurable*.<br>
+
+When I see problems and workarounds like that it makes me wonder, what is here the
+benefit of using Gradle? Doesn't it just give people more rope to hang themselves?<br>
```

Comment:
Add a smiley

Reply:
A smiley would not fit the general tone I think. Besides, I am dead serious ;)